pyFlowSOM/cyFlowSOM.pyx

```diff
@@ -142,8 +143,8 @@ def som(
     xDists = np.zeros(n_nodes, dtype=np.dtype("d"))
     cdef double [:] xDists_mv = xDists
 
-    if deterministic == True:
-        C_SEED_RAND(2407230991)
+    if seed is not None:
+        C_SEED_RAND(seed)
```

Comment:
In the description, you wrote if seed is None, then it runs with fully stochastic node initialization - but is that true? It seems like if seed is None, then `C_SEED_RAND` isn't called at all. From my very limited testing, it seems like c chooses some reproducible seed if  `C_SEED_RAND` isn't set (but I could be wrong about this).

I think to be truly stochastic, we should change this to:
```
if seed is not None:
    C_SEED_RAND(seed)
else:
    C_SEED_RAND(np.random.randint(low=1, high=65535)
```

or something like that.

Reply:
Just to be safe I think it's a good idea to add this. It may be a machine-dependent issue of how randomization is set.

On a side note, we haven't made the change in `ark-analysis` yet to support a `seed` parameter for `train_pixel_som` or `train_cell_som`. That will require a separate PR; right now, it will always use `seed=42` as the default argument provided to `pyFlowSOM.som`.